src/PVRFilmonData.cpp

```diff
@@ -280,60 +365,57 @@ int PVRFilmonData::UpdateChannel(unsigned int channelId)
-                                          int iChannelUid,
-                                          time_t iStart,
-                                          time_t iEnd)
+PVR_ERROR PVRFilmonData::GetEPGForChannel(int channelUid, time_t start, time_t end,
+                                          kodi::addon::PVREPGTagsResultSet& results)
 {
   std::lock_guard<std::mutex> lock(m_mutex);
-  XBMC->Log(LOG_DEBUG, "getting EPG for channel");
-  unsigned int broadcastIdCount = lastTimeChannels;
-  int chIndex = PVRFilmonData::UpdateChannel(iChannelUid);
+  kodi::Log(ADDON_LOG_DEBUG, "getting EPG for channel");
+  unsigned int broadcastIdCount = m_lastTimeChannels;
+  int chIndex = PVRFilmonData::UpdateChannel(channelUid);
   if (chIndex >= 0)
   {
     PVRFilmonChannel ch = m_channels[chIndex];
     for (unsigned int epgId = 0; epgId < ch.epg.size(); epgId++)
     {
       PVRFilmonEpgEntry& epgEntry = ch.epg.at(epgId);
-      if (epgEntry.startTime >= iStart && epgEntry.endTime <= iEnd)
+      if (epgEntry.startTime >= start && epgEntry.endTime <= end)
       {
-        EPG_TAG tag;
-        memset(&tag, 0, sizeof(EPG_TAG));
-        tag.iUniqueBroadcastId = broadcastIdCount++;
-        tag.strTitle = epgEntry.strTitle.c_str();
-        tag.iUniqueChannelId = epgEntry.iChannelId;
-        tag.startTime = epgEntry.startTime;
-        tag.endTime = epgEntry.endTime;
-        tag.strPlotOutline = epgEntry.strPlotOutline.c_str();
-        tag.strPlot = epgEntry.strPlot.c_str();
-        tag.strOriginalTitle = nullptr; // unused
-        tag.strCast = nullptr; // unused
-        tag.strDirector = nullptr; // unused
-        tag.strWriter = nullptr; // unused
-        tag.iYear = 0; // unused
-        tag.strIMDBNumber = nullptr; // unused
-        tag.strIconPath = epgEntry.strIconPath.c_str();
-        tag.iGenreType = epgEntry.iGenreType;
-        tag.iGenreSubType = epgEntry.iGenreSubType;
-        tag.strGenreDescription = "";
-        tag.strFirstAired = "";
-        tag.iParentalRating = 0;
-        tag.iStarRating = 0;
-        tag.iSeriesNumber = EPG_TAG_INVALID_SERIES_EPISODE;
-        tag.iEpisodeNumber = EPG_TAG_INVALID_SERIES_EPISODE;
-        tag.iEpisodePartNumber = EPG_TAG_INVALID_SERIES_EPISODE;
-        tag.strEpisodeName = "";
-        tag.iFlags = EPG_TAG_FLAG_UNDEFINED;
-        PVR->TransferEpgEntry(handle, &tag);
+        kodi::addon::PVREPGTag tag;
+        tag.SetUniqueBroadcastId(broadcastIdCount++);
+        tag.SetTitle(epgEntry.strTitle);
+        tag.SetUniqueChannelId(epgEntry.iChannelId);
+        tag.SetStartTime(epgEntry.startTime);
+        tag.SetEndTime(epgEntry.endTime);
+        tag.SetPlotOutline(epgEntry.strPlotOutline);
+        tag.SetPlot(epgEntry.strPlot);
+        tag.SetOriginalTitle(""); // unused
+        tag.SetCast(""); // unused
+        tag.SetDirector(""); // unused
+        tag.SetWriter(""); // unused
+        tag.SetYear(0); // unused
+        tag.SetIMDBNumber(""); // unused
```

Comment:
Note about this.
The set here and on some other places are not needed and can be also leaved away. 
Only added to have equal to before and to become allow for remove. 

`+++` We see the used parts
`---` Some not needed work

What think the Team remove or leave?